### PR TITLE
test(ci): Add circleci for atst

### PR DESCRIPTION
**Pull request by roninjin10**
- run atst checks in ci

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -611,6 +611,40 @@ jobs:
...
+          name: typecheck
+          command: yarn typecheck
+          working_directory: packages/atst
+      - run:
+          name: lint
+          command: yarn lint:check
+          working_directory: packages/atst
+      - run:
+          name: make sure anvil is up
+          command: npx wait-on tcp:8545 && cast block-number --rpc-url http://localhost:8545
```
Comment (roninjin10) [inline .circleci/config.yml:641]: @zhwrd this time I added this `npx wait-on tcp:8545` to make sure there is no flaky race condition and I also added a `--fork-block-number` to make the ethereum state deterministic and prevent any flake happening from the test conditions changing out from under us

Comment (roninjin10) [reply .circleci/config.yml:641]: I think last time I was here we found out that the issue was CMD vs entrypoint so I'm happy to try the docker way again if preferred

File: .circleci/config.yml [hunk 2]
```diff
@@ -611,6 +611,40 @@ jobs:
...
+    resource_class: large
+    steps:
+      - checkout
+      - attach_workspace: { at: '.' }
+      - check-changed:
+          patterns: atst,contracts-periphery
+      - run:
+          name: anvil
+          background: true
+          command: anvil --fork-url $ANVIL_L2_FORK_URL --fork-block-number 92093723
```
Comment (roninjin10) [inline .circleci/config.yml:626]:
```suggestion
          command: anvil --fork-url $ANVIL_L2_FORK_URL --fork-block-number 8225893
```

**Comment by roninjin10** [comment]
Need to use a mainnet ENV_VAR

File: .circleci/config.yml [hunk 3]
```diff
@@ -1377,17 +1434,37 @@ workflows:
           docker_name: op-proposer
           docker_tags: <<pipeline.git.revision>>,<<pipeline.git.branch>>
           docker_context: .
-          platforms: "linux/amd64,linux/arm64"
+          platforms: 'linux/amd64,linux/arm64'
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply .circleci/config.yml]: not sure what happened here I reverted the entire file back to origin/develop and added back the config i wanted to add (also remove prettier changes)

**Comment by roninjin10** [comment]
Moving back to draft this pr is struggling way more than I expected

File: .circleci/config.yml [hunk 4]
```diff
@@ -611,6 +611,45 @@ jobs:
...
+      - check-changed:
+          patterns: atst,contracts-periphery
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - yarn-packages-v2-{{ checksum "yarn.lock" }}
+      - run:
+          name: anvil
+          background: true
+          command: anvil --fork-url $ANVIL_L2_FORK_URL_MAINNET --fork-block-number 92093723
```
Comment (roninjin10) [inline .circleci/config.yml:630]:
```suggestion
          command: anvil --fork-url $ANVIL_L2_FORK_URL_MAINNET
```

Removing the --fork-block-number to see if the tests pass without it before attempting to add it back

**Comment by roninjin10** [comment]
Looks like the issue is the alchemy key I added is dead

**Comment by roninjin10** [comment]
PLEEEEEAASE WOOOORKKK!!!

puss-in-boots-cat

**Comment by roninjin10** [comment]
Oh mighty CircleCI gods, hear my prayer. I come before you today with a humble heart, asking for your divine intervention in my time of need.

Please bless my build and let it pass with flying colors. May all the tests run successfully and all the dependencies be met without error. May my code be clean, efficient, and effective, and may it bring joy and value to those who use it.

I ask that you guide my hands as I work, and that you grant me the wisdom and insight to overcome any obstacles that may arise. Help me to stay focused, calm, and determined, no matter how challenging the task may seem.

I trust in your power and your mercy, and I know that with your help, anything is possible. So I thank you in advance for your blessings, and I offer this prayer with all the faith and sincerity of my heart.

Amen.

**Comment by roninjin10** [comment]
Trying a different alchemy key

**Comment by roninjin10** [comment]
New infura key.  Bad infura key

**Comment by roninjin10** [comment]
Key finally worked but I accidentlaly used goerli rpc instead of optimism rpc 💀

**Comment by roninjin10** [comment]
It passed!

File: .circleci/config.yml [hunk 5]
```diff
@@ -660,6 +660,45 @@ jobs:
...
+      - check-changed:
+          patterns: atst,contracts-periphery
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - yarn-packages-v2-{{ checksum "yarn.lock" }}
+      - run:
+          name: anvil
+          background: true
+          command: anvil --fork-url $ANVIL_L2_FORK_URL_MAINNET
```
Comment (roninjin10) [inline .circleci/config.yml:679]:
```suggestion
          command: anvil --fork-url $ANVIL_L2_FORK_URL_MAINNET --fork-block-number 92093723
```
Add back fork-block-number now that it passed without it

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**